latest-release.ts

```diff
@@ -0,0 +1,31 @@
+import * as nodegit from "nodegit";
+import * as semver from "semver";
+
+
+export async function checkoutLatestRelease(
+    r: nodegit.Repository|Promise<nodegit.Repository>, dir?: string)
+        : Promise<nodegit.Repository> {
+  const repo = await r;
+  const repos = await nodegit.Tag.list(repo);
+  let latestRelease: string;
+  if (repos && repos.length > 0) {
+    try {
+      latestRelease = repos.sort(semver.rcompare)[0];
+    } catch (err) {
+      // We couldn't pick a release, so we'll just bow out here.
+      return repo;
+    }
+  } else {
+    return repo;
+  }
+  let commit: nodegit.Commit;
+  try {
+    commit = await repo.getReferenceCommit(latestRelease);
```

Comment:
If it can't find a commit for that release for any reason it throws.

Why anything?